Adding: IDP Tutorial

JensHaley opened the pull request:
**What this PR does / why we need it**:

Adding tutorial about how to use an OIDC compliant IDP provider to authenticate users to shoot clusters.

**Special notes for your reviewer**:

I revised what we had to "dust it off". 

After putting some thought into it I thought `/documentation` would be a good home as the tutorial is a cross-topic.

JensHaley commented:
/invite @mvladev - a few changes were made to the original version, I'd appreciate another pair of eyes to confirm my changes

mvladev commented on website/documentation/tutorials/oidc-login/_index.md:
````diff
@@ -0,0 +1,209 @@
+
+1. Install the `kubectl` plugin [oidc-login](https://github.com/int128/kubelogin). We highly recommend the [krew](https://github.com/kubernetes-sigs/krew) install tool, which also makes other plugins easily available.
+
+    ```
````
use 

```suggestion
    ```console
```

to mark a [shell console](https://github.com/github/linguist/blob/633d196c170e002bcccb4143d6906787f360e08b/lib/linguist/languages.yml#L5251-L5262).

mvladev commented on website/documentation/tutorials/oidc-login/_index.md:
````diff
@@ -0,0 +1,209 @@
+
+2. Prepare a `kubeconfig` for later use:
+   
+    ```
````
use 

```suggestion
    ```console
```

to mark a [shell console](https://github.com/github/linguist/blob/633d196c170e002bcccb4143d6906787f360e08b/lib/linguist/languages.yml#L5251-L5262).

mvladev commented on website/documentation/tutorials/oidc-login/_index.md:
````diff
@@ -0,0 +1,209 @@
+        usernameClaim: email
+```
+
+This change of the `Shoot` manifest triggers a reconciliation. Once the reconciliation is finished, your OIDC configuration is applied. It does *not* invalidate other certificate based authentication methods. Wait for Gardener to reconcile the change. It can take up to 5 minutes.
````
```suggestion
This change of the `Shoot` manifest triggers a reconciliation. Once the reconciliation is finished, your OIDC configuration is applied. It does *not* invalidate other certificate-based authentication methods. Wait for Gardener to reconcile the change. It can take up to 5 minutes.
```

mvladev commented on website/documentation/tutorials/oidc-login/_index.md:
```diff
@@ -0,0 +1,209 @@
+    ...
+
+    users:
+    - name: my-oidc
+      user:
+        auth-provider:
+          config:
+            client-id: <Client ID>
+            client-secret: <Client Secret>
+            idp-issuer-url: "https://<Issuer>/"
+            extra-scopes: email,offline_access,profile
+          name: oidc
```
Use the new exec plugin:

```suggestion
    - name: my-oidc
      user:
        exec:
          apiVersion: client.authentication.k8s.io/v1beta1
          command: kubectl
          args:
          - oidc-login
          - get-token
          - --oidc-issuer-url=https://<Issuer>/ 
          - --oidc-client-id=<Client ID>
          - --oidc-client-secret=<Client Secret>
          - --oidc-extra-scope=email,offline_access,profile
```

mvladev commented on website/documentation/tutorials/oidc-login/_index.md:
````diff
@@ -0,0 +1,209 @@
+
+## Verify the Result
+
+1. Activate the prepared `kubeconfig-oidc` and perform a login:
+   
+    ```
+    $ export KUBECONFIG=~/.kube/config-oidc
+    $ kubectl oidc-login
+    Open http://localhost:8000 for authentication
+    ```
+
+    The plugin opens a browser for an interctive authentication session, and in parallel serves a local webserver for the configured callback.
+
+    ![login](./login.png)
+
+2. Enter your login credentials. If you successfully verified your user the console will display the validity of your returned token:
+  
+    ```
+    You got a valid token until 2019-08-14 06:26:49 +0200 CEST
+    ```
+
+3. Inspect the `kubeconfig-oidc`. You will find two additional parameters:
+   
+    ```yaml
+    ...
+    users:
+    - name: my-oidc
+      user:
+        auth-provider:
+          config:
+            client-id: <Client ID>
+            client-secret: <Client Secret>
+            idp-issuer-url: "https://<Issuer>/"
+            extra-scopes: email,offline_access,profile
+            id-token: eyJ0eX ... 4In0.QQKS ... TTTw
+            refresh-token: LFt ... 0Skj
+          name: oidc
+    ```
+
+    The plugin persisted the `id-token` and `refresh-token` in your configuration file. 
````
Those steps are not needed / could be rewritten. With the newest version of [kubelogin](https://github.com/int128/kubelogin#setup) you can do `kubectl get pods` and then you are automatically authenticated in the browser if you haven't been authenticated yet.

mvladev reviewed (changes requested):
Some changes for the newest kubelogin version.

JensHaley commented:
@mvladev : Thanks! 

I went through the whole tutorial again to include your changes and to rewrite the verification step:
* I didn't like to use cluster role `cluster-admin`; I think it's easier to notice the difference in the end using cluster role `view`
* `kubectl who-can` didn't work for me - users need another plugin for this command. I used `kubectl auth can-i` instead
* I added a few more up-to-date screenshots, and added more explanation, because I got stuck on minor things

How does it look from your point of view now?

mvladev reviewed (approved):
/lgtm
